app/src/main/java/com/example/poste/models/User.java

```diff
@@ -302,4 +304,62 @@ private String parseError(Response<ResponseBody> response) {
         }
         return null;
     }
+
+    /**
+     * Adds a new post the given folder
+     *
+     * @param folder folder to add the new post to
+     */
+    public void createNewPost(Folder folder, Context context, String title, String description, String url) {
+        MyApiService apiService = RetrofitClient.getRetrofitInstance().create(MyApiService.class);
+        Call<ResponseBody> call = apiService.createPost(getTokenHeader() ,new CreatePost(title, description, url, folder.getTitle()));
+        call.enqueue(new Callback<ResponseBody>() {
+            @Override
+            public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
+                if (response.isSuccessful()) {
+                    updateFoldersAndPosts(context);
+                }
+                else {
+                    try {
+                        Toast.makeText(context, response.errorBody().string(), Toast.LENGTH_LONG);
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+            }
+            @Override
+            public void onFailure(Call<ResponseBody> call, Throwable t) {
+                Toast.makeText(context, "Server Not Responding", Toast.LENGTH_LONG);
+            }
+        });
+    }
+
+    /**
+     * Deletes a given post from a given folder in the data model and backend
+     *
+     * @param post the post you want to delete
+     */
+    public void deletePost(Context context, Post post, Folder folder) {
```

Comment:
I thought I mentioned; if not, I apologize. The PR I put up a few days ago has a DeletePost implementation already, with the callback we discussed in the huddle this morning. Your choice how you want to press forward. If you want to steal that implementation, feel free to take a gander at it.

Reply:
once we get all the PR's together into dev I will mimic your callback work and aptly it to all API calls for the MVP.

Reply:
It might be more efficient if we merge yours in first and I resolve the conflicts on my branch, since it will take some work integrating the callbacks, etc.